src/systolic_array/commit.cpp

```diff
@@ -68,9 +68,17 @@ void Commit::evaluate() {
   // PE row is left idle. 2) Some of the PE columns are left idle due to a lack
   // of weights. In this case, we should do a writeback once all the "active"
   // columns have produced outputs.
-  for (int i = 0; i < ceil(float(inputs.size()) / elemsPerLine); i++) {
-    for (int j = 0; j < elemsPerLine && j < inputs.size(); j++) {
-      int index = i * elemsPerLine + j;
+
+  for (int remainingElems = inputs.size(); remainingElems > 0;
+       remainingElems -= elemsPerLine) {
+    // Check if we have collected all the pixels for a writeback.
+    int elemsToWrite = std::min(elemsPerLine, remainingElems);
+    int index = inputs.size() - remainingElems;
```

Comment:
I'd pick a more descriptive name than "index". Something that tells me it represents the "start" of something would be better (based on how it's used in queueCommitRequest).

Reply:
Yes. Changed the name to `start`.